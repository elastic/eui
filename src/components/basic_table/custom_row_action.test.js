import React from 'react';
import { shallow } from 'enzyme/build/index';
import { CustomRowAction } from './custom_row_action';

describe('CustomRowAction', () => {

  test('render', () => {

    const props = {
      action: {
        name: 'custom1',
        description: 'custom 1',
        render: () => 'test'
      },
      enabled: true,
      visible: true,
      row: { id: 'xyz' },
      model: {
        data: {
          rows: [],
          totalRowCount: 0
        },
        criteria: {
          page: {
            size: 5,
            index: 0
          }
        }
      }
    };

    const component = shallow(
      <CustomRowAction {...props} />
    );

    expect(component).toMatchSnapshot();

  });

});
