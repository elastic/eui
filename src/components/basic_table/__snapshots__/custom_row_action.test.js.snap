// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRowAction render 1`] = `
<div
  style={
    Object {
      "opacity": 1,
    }
  }
/>
`;
